Add exec output-schema parameter

pakrym-oai opened the pull request:
Adds structured output to `exec` via the `--structured-output` parameter.

pakrym-oai committed:
Add notifier tests

pakrym-oai committed:
Merge branch 'main' into pakrym/add-notifier-tests

# Conflicts:
#	codex-rs/core/tests/common/Cargo.toml

pakrym-oai committed:
codex-rs/core: remove obsolete notify.txt file

pakrym-oai committed:
user_notification: skip test on Windows and refactor sandbox check with macro

pakrym-oai committed:
Add exec output-schema parameter

pakrym-oai committed:
README.md: remove outdated section about GPT-5 structured JSON output and schema validation

pakrym-oai committed:
fb

aibrahim-oai reviewed (approved):
There are some unrelated changes that would be best in a different PR. Some comments about the test assertion. Beside this, awesome work!

pakrym-oai committed:
Merge branch 'main' into pakrym/add-notifier-tests

pakrym-oai committed:
Merge branch 'pakrym/add-notifier-tests' into pakrym/add-exec-output-schema-parameter

# Conflicts:
#	codex-rs/core/src/codex/compact.rs

pakrym-oai committed:
core/client_common.rs: simplify verbosity conversion with Into::into
exec/tests/suite/output_schema.rs: refactor to build schema with serde_json, write as pretty json, and assert full format object equality
